proofs/pfedit.ml

```diff
@@ -37,46 +37,35 @@ let get_nth_V82_goal p i =
   try { it = List.nth goals (i-1) ; sigma }
   with Failure _ -> raise NoSuchGoal
 
-let get_goal_context_gen p i =
-  let { it=goal ; sigma=sigma; } = get_nth_V82_goal p i in
+let get_goal_context_gen pf i =
+  let { it=goal ; sigma=sigma; } = get_nth_V82_goal pf i in
   (sigma, Refiner.pf_env { it=goal ; sigma=sigma; })
 
-let get_goal_context i =
-  try get_goal_context_gen (Proof_global.give_me_the_proof ()) i
-  with Proof_global.NoCurrentProof -> CErrors.user_err Pp.(str "No focused proof.")
-     | NoSuchGoal -> CErrors.user_err Pp.(str "No such goal.")
-
-let get_current_goal_context () =
-  try get_goal_context_gen (Proof_global.give_me_the_proof ()) 1
-  with Proof_global.NoCurrentProof -> CErrors.user_err Pp.(str "No focused proof.")
-     | NoSuchGoal -> 
-    (* spiwack: returning empty evar_map, since if there is no goal, under focus,
-        there is no accessible evar either *)
-    let env = Global.env () in
-    (Evd.from_env env, env)
+let get_goal_context pf i =
+  let p = Proof_global.give_me_the_proof pf in
+  get_goal_context_gen p i
 
-let get_current_context ?p () =
-  let current_proof_by_default = function
-    | Some p -> p
-    | None -> Proof_global.give_me_the_proof ()
-  in
-  try get_goal_context_gen (current_proof_by_default p) 1
-  with Proof_global.NoCurrentProof ->
+let get_current_goal_context pf =
+  let p = Proof_global.give_me_the_proof pf in
+  try get_goal_context_gen p 1
+  with
+  | NoSuchGoal ->
+    (* spiwack: returning empty evar_map, since if there is no goal,
+       under focus, there is no accessible evar either. EJGA: this
+       seems strange, as we have pf *)
     let env = Global.env () in
-    (Evd.from_env env, env)
-     | NoSuchGoal ->
-        (* No more focused goals ? *)
-        let p = (current_proof_by_default p) in
-        let evd = Proof.in_proof p (fun x -> x) in
-        (evd, Global.env ())
-
-let current_proof_statement () =
-  match Proof_global.V82.get_current_initial_conclusions () with
-    | (id,([concl],strength)) -> id,strength,concl
-    | _ -> CErrors.anomaly ~label:"Pfedit.current_proof_statement" (Pp.str "more than one statement.")
+    Evd.from_env env, env
+
+let get_current_context pf =
+  let p = Proof_global.give_me_the_proof pf in
+  try get_goal_context_gen p 1
+  with
+  | NoSuchGoal ->
+    (* No more focused goals *)
+    let evd = Proof.in_proof p (fun x -> x) in
+    evd, Global.env ()
 
 let solve ?with_end_tac gi info_lvl tac pr =
-  try 
```

Comment:
Why was this handler removed? It may be related to abstract, so I'd like to understand better.

Reply:
The handler doesn't make sense anymore, as typing guarantees this cannot be called wihtout a proof now.